controller/lib/serdes/controller_data_packet_encoder.cpp

```diff
@@ -0,0 +1,30 @@
+#include "serdes.h"
```

Comment:
Perhaps merge these two cpp files into simply `lib/core/serdes.cpp`?  They're not big, so maybe a little easier to understand  if the relevant code is all next to each other.